terraform/main.tf

```diff
@@ -84,7 +84,10 @@ resource "null_resource" "apply_deployment" {
 resource "null_resource" "wait_conditions" {
   provisioner "local-exec" {
     interpreter = ["bash", "-exc"]
-    command     = "kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=-1s"
+    command     = <<-EOT
+    kubectl wait --for=condition=AVAILABLE apiservice/v1beta1.metrics.k8s.io --timeout=1200s
+    kubectl wait --for=condition=ready pods --all -n ${var.namespace} --timeout=1200s
+    EOT
```

Comment:
The previously used `--timeout` of `-1s` refers to 1 week. [Related docs here](https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#wait:~:text=negative%20means%20wait%20for%20a%20week.).